src/concatAll.js

```diff
@@ -0,0 +1,34 @@
+import { concat, identical, identity, pipe, when, reduce } from 'ramda';
+
+import stubNull from './stubNull';
+import fl from './fantasy-land/mapping';
+
+const nullSemigroup = { [fl.concat]: identity };
+
+/**
+ * Returns the result of concatenating all semigroups (such as arrays or strings) in passed foldable (such as an array).
```

Comment:
```
Returns the result of concatenating the given lists or strings.
Note: R.concat expects all items to be of the same type. It will throw an error if you concat an Array with a non-Array value.
Dispatches to the concat method of the item, if present. Can also concatenate multiple members of a fantasy-land compatible semigroup.
```

Reply:
We should also add info about returning undefined. It's missing in this description. Please do so.